### DISCUSSION TITLE
Scalar affecting Docusaurus CSS configuration

### DISCUSSION THREAD

**Discussion by htessaro** [scalar/scalar#2362] [Q&A] [answered]
I've added the Scalar plugin to a Docusaurus Project. The plugin is working fine. However, after adding the plugin to the project through `docusaurus.config.js`, I noticed that existing custom CSS configurations were affected. For example, the codeblocks were changed entirely in pages outside of the API Reference page generated by the Scalar Plugin. 

Is there some configuration that I should do to keep the Scalar CSS configuration from messing up the other existing Docusaurus pages?

Below, you find examples comparing before and after adding the Scalar Plugin to the project:

Before:
![image](https://github.com/scalar/scalar/assets/85108199/d4d58dbb-7dc7-458e-ad01-fe1a93102727)

After:
![image](https://github.com/scalar/scalar/assets/85108199/1391eae4-791b-4d9a-b280-6a92901ae4e5)

**Comment by amritk** [comment]
hey @heitortessaro do you think you could inspect and see which styles exactly are overwriting?

Thanks

**Comment by htessaro** [comment]
Here's the docusaurus.config.js configuration I used:

```
plugins: [
    require.resolve('docusaurus-lunr-search'),
    require.resolve('docusaurus-plugin-image-zoom'),
    [
      '@scalar/docusaurus',
      {
        label: 'API Reference',
        route: '/api',
        configuration: {
          hideModels: true,
          spec: {
            // Put the URL to your OpenAPI document here:
            url: 'https://raw.githubusercontent.com/bucketeer-io/bucketeer/003d33b7214c06305cefc00ce62cc0c1be1f754d/api-description/apidocs.swagger.yaml'
          },
        },
      },
    ],
  ],
```

**Comment by htessaro** [comment]
@amritk I was checking the CSS classes in the codebook, and I don't see many differences. Below you find the CSS classes of the div, pre, and code tags, which create the codeblock share in the previous message.

External div:

- Without the plugin: `codeBlockContent_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module`
- With the plugin:      `codeBlockContent_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module`

pre: 

- Without: `prism-code language-swift codeBlock_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module thin-scrollbar`
- With:       `prism-code language-swift codeBlock_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module thin-scrollbar`

Code:

- Without: `codeBlockLines_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module codeBlockLinesWithNumbering_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module`
- With: `codeBlockLines_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module codeBlockLinesWithNumbering_node_modules-@docusaurus-theme-classic-lib-theme-CodeBlock-Content-styles-module`

**Comment by hwkr** [comment]
I have a feeling this is also an issue with the tailwind preflight coming in from `@scalar/api-client`. I'm guessing the issue you're seeing is actually [this line targeting `pre`](https://github.com/tailwindlabs/tailwindcss/blob/3.3/src/css/preflight.css#L288).

All going well #2363 should fix it.

**Reply by htessaro** [reply]
Thanks @hwkr

**Reply by htessaro** [reply]
Hi @hwkr, I noticed #2363 was released. However, after updating the plugin, I continue to face the same leakage CSS problem as reported before. Is there a way to encapsulate all Scalar CSS configurations to ensure they won't affect other pages in the documentation?

**Reply by hwkr** [reply]
I think I may have fixed the general case but not docusaurus, I'll have a look today!

**Reply by niwsa** [reply]
https://github.com/scalar/scalar/blob/main/packages/docusaurus/src/theme.css#L678 is setting the min-width to 100px instead of 100% (which was the earlier value). Also affected are the bg-color and font styles.
<img width="950" alt="Screenshot 2024-07-10 at 4 40 11 PM" src="https://github.com/scalar/scalar/assets/3107922/1f68e123-3fb2-4d0a-a35a-022dcdfa6a36">

See Working preview of code block (v0.4.13 of plugin: https://f6fb1187.website-d9q.pages.dev/docs/jackson/sso-flow#21-add-connection)

**Reply by hwkr** [reply]
Fix is in #2454 :)

**Reply by htessaro** [reply]
Thanks @hwkr, now all is working 😄 😁

**Reply by htessaro** [reply]
@hwkr. I was testing the project after updating the scalar plugin to v0.4.49 and the project works normally when I run in local mode. However, when I built the Docusaurus project, the same problems appeared. I noticed that after updating the Scalar plugin when I built the project, I received the following warning messages, which I had never faced before:

```
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nUnexpected '}' at assets/css/styles.b3c4664c.css:43:9723.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nUnexpected '}' at assets/css/styles.b3c4664c.css:43:9813.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nUnexpected '}' at assets/css/styles.b3c4664c.css:43:10243.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nUnexpected '}' at assets/css/styles.b3c4664c.css:43:10418.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid property name 'code.hljs *{text-align' at assets/css/styles.b3c4664c.css:43:2960. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid property name '.introduction-card-row[data-v-0b17f72e]>*{min-width' at assets/css/styles.b3c4664c.css:43:9749. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid property name '.algolia-autocomplete.algolia-autocomplete-right .ds-dropdown-menu{left' at assets/css/styles.b3c4664c.css:43:9839. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid property name '.algolia-docsearch-suggestion{border-bottom-color' at assets/css/styles.b3c4664c.css:43:10269. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid property name '' at assets/css/styles.b3c4664c.css:43:10444. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:800px)' at assets/css/styles.b3c4664c.css:43:10418. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:768px)' at assets/css/styles.b3c4664c.css:43:10243. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:600px)' at assets/css/styles.b3c4664c.css:43:9723. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:800px)' at assets/css/styles.b3c4664c.css:43:10418. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:768px)' at assets/css/styles.b3c4664c.css:43:10243. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:600px)' at assets/css/styles.b3c4664c.css:43:9723. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:800px)' at assets/css/styles.b3c4664c.css:43:10418. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:768px)' at assets/css/styles.b3c4664c.css:43:10243. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:600px)' at assets/css/styles.b3c4664c.css:43:9723. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:800px)' at assets/css/styles.b3c4664c.css:43:10418. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:768px)' at assets/css/styles.b3c4664c.css:43:10243. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:600px)' at assets/css/styles.b3c4664c.css:43:9723. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:800px)' at assets/css/styles.b3c4664c.css:43:10418. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:768px)' at assets/css/styles.b3c4664c.css:43:10243. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:600px)' at assets/css/styles.b3c4664c.css:43:9723. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:800px)' at assets/css/styles.b3c4664c.css:43:10418. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:768px)' at assets/css/styles.b3c4664c.css:43:10243. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:600px)' at assets/css/styles.b3c4664c.css:43:9723. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:800px)' at assets/css/styles.b3c4664c.css:43:10418. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:768px)' at assets/css/styles.b3c4664c.css:43:10243. Ignoring.","compilerPath":"client"}
[WARNING] {"message":"assets/css/styles.b3c4664c.css from Css Minimizer plugin\nInvalid selector '}@media (min-width:600px)' at assets/css/styles.b3c4664c.css:43:9723. Ignoring.","compilerPath":"client"}
```

**Reply by niwsa** [reply]
We have the same issue in production 
 https://github.com/scalar/scalar/pull/2454#issuecomment-2226889437

**Reply by htessaro** [reply]
Got it. However, you're not facing any CSS issues, correct?

**Reply by deepakprabhakara** [reply]
No specific CSS issue per se but the scalar CSS is still conflicting with Docusaurus code block theme in the production build. So in effect the fix here works locally but not in the production build.

**Reply by hanspagel** [reply]
Looks like nested styles cause the issue:

```css
.scalar-app {
  code.hljs * {
    text-align: left;
  }
}
```

This should be widely supported in browsers, but probably isn’t in the used CSS minifier. We should probably do some PostCSS processing here. Wdyt @hwkr?

**Reply by hwkr** [reply]
Yeah PostCSS is probably the way to go, I'll have a look.

**Reply by hwkr** [reply]
I have a fix in review in #2546 :)

**Reply by htessaro** [reply, marked as answer]
Thank you @hwkr . I just tested the[ 0.4.52](https://www.npmjs.com/package/@scalar/docusaurus/v/0.4.52) version. Now, there are no leaks in the CSS. I'll close the discussion.

**Reply by deepakprabhakara** [reply]
Confirmed, we don't see any issues now. Thanks @hwkr :)
